Refactoring doc to better reflect DogStatsD funct (#473)
DataDog/datadogpy: README.md

# The Datadog Python library

[![Unit Tests](https://dev.azure.com/datadoghq/datadogpy/_apis/build/status/DataDog.datadogpy.unit?branchName=master)](https://dev.azure.com/datadoghq/datadogpy/_build/latest?definitionId=10&branchName=master)
[![Integration Tests](https://dev.azure.com/datadoghq/datadogpy/_apis/build/status/DataDog.datadogpy.integration?branchName=master)](https://dev.azure.com/datadoghq/datadogpy/_build/latest?definitionId=13&branchName=master)
[![Documentation Status](https://readthedocs.org/projects/datadogpy/badge/?version=latest)](https://readthedocs.org/projects/datadogpy/?badge=latest)
[![PyPI - Version](https://img.shields.io/pypi/v/datadog.svg)](https://pypi.org/project/datadog)
[![PyPI - Downloads](https://pepy.tech/badge/datadog)](https://pepy.tech/project/datadog)

> Datadogpy is a collection of tools suitable for inclusion in existing Python projects or for development of standalone scripts. It provides an abstraction on top of Datadog's raw HTTP interface and the Agent's DogStatsD metrics aggregation server, to interact with Datadog and efficiently report events and metrics.

Reviewer: ```suggestion
The Datadog Python Library is a collection of tools suitable for inclusion in existing Python projects or for the development of standalone scripts. It provides an abstraction on top of Datadog's raw HTTP interface and the Agent's DogStatsD metrics aggregation server, to interact with Datadog and efficiently report events and metrics.
```

    python setup.py install

## Datadog API

> Find below a working example to submit an Event to your Event Stream:

Reviewer: ```suggestion
Find below a working example for submitting an event to your Event Stream:
```

```python
from datadog import initialize, api

options = {
    'api_key': '<YOUR_API_KEY>',
    'app_key': '<YOUR_APP_KEY>'
}

initialize(**options)

title = "Something big happened!"
text = 'And let me tell you all about it here!'
tags = ['version:1', 'application:web']

api.Event.create(title=title, text=text, tags=tags)
```

> **Consult the full list of supported Datadog API endpoint with working code examples in [Datadog-API documentation](https://docs.datadoghq.com/api/?lang=python).**

Reviewer: ```suggestion
**Consult the full list of supported Datadog API endpoints with working code examples in [the Datadog API documentation](https://docs.datadoghq.com/api/?lang=python).**
```

> **Note**: The full list of available Datadog API endpoint is also available in the [Datadog-py Readthedoc documentation](https://datadogpy.readthedocs.io/en/latest/)

Reviewer: ```suggestion
**Note**: The full list of available Datadog API endpoints is also available in the [Datadog Python Library documentation](https://datadogpy.readthedocs.io/en/latest/)
```

api.Event.create(title=title, text=text, tags=tags)
```

## DogStatsD

In order to use DogStatsD metrics, the Agent must be [running and available](https://docs.datadoghq.com/developers/dogstatsd/?tab=python).

### Instantiate the DogStatsD client

> Once Datadog-py is installed, instantiate it in your code:

Reviewer: ```suggestion
Once the Datadog Python Library is installed, instantiate the StatsD client in your code:
```

```python
from datadog import statsd

options = {
    'statsd_host':'127.0.0.1',
    'statsd_port':8125
}

> initialize(**options)

Author: 🤔 we could indeed, but it's in case people won't click on the link. At least they have the minimum configuration required for it to work.

It's also to avoid this page to be just a summary of links which could create a "Wall of text" effect i think.

> initialize(**options)

Reviewer 2: ok :+1:

```

> See the full list of available [DogStatsD Client instantiation parameters](https://docs.datadoghq.com/developers/dogstatsd/?tab=python#client-instantiation-parameters).

Reviewer: ```suggestion
See the full list of available [DogStatsD client instantiation parameters](https://docs.datadoghq.com/developers/dogstatsd/?tab=python#client-instantiation-parameters).
```

#### Origin detection over UDP

Origin detection is a method to detect which pod `DogStatsD` packets are coming from in order to add the pod's tags to the tag list.

> The `DogStatsD` client attaches an internal tag, `entity_id`. The value of this tag is the content of the `DD_ENTITY_ID` environment variable if found, which is the pod's UID. This tag is used by the Datadog Agent to insert container tags to the metrics. You should only `append` to the `constant_tags` list to avoid overwriting this global tag.

Reviewer: ```suggestion
The `DogStatsD` client attaches an internal tag, `entity_id`. The value of this tag is the content of the `DD_ENTITY_ID` environment variable if found, which is the pod's UID. The Datadog Agent uses this tag to add container tags to the metrics. To avoid overwriting this global tag, make sure to only `append` to the `constant_tags` list.
```
